internal/dataplane/translator/translate_httproute_test.go

```diff
@@ -1758,34 +1746,23 @@ func TestIngressRulesFromHTTPRoutes_RegexPrefix(t *testing.T) {
 			return func(t *testing.T) {
 				ingressRules := newIngressRules()
 
-				var errs []error
 				for _, httproute := range tt.routes {
 					// initialize the HTTPRoute object
 					httproute.SetGroupVersionKind(httprouteGVK)
-
-					// generate the ingress rules
-					err := tran.ingressRulesFromHTTPRoute(&ingressRules, httproute)
-					if err != nil {
-						errs = append(errs, err)
-					}
 				}
+				tran.ingressRulesFromHTTPRoutesWithCombinedService(tt.routes, &ingressRules)
```

Comment:
As I understand, `TestIngressRulesFromHTTPRoutes` now covers only the cases with `CombinedServicesFromDifferentHTTPRoutes=false`. I'm not sure about what is our testing strategy here, but shouldn't we also cover the case with `=true` on this layer?

Reply:
Added another set of test cases for `CombinedServicesFromDifferentHTTPRoutes=true`.